ee/app/message/service.ts

```diff
@@ -40,7 +40,12 @@ export class MessageEnterprise extends ServiceClass implements IMessageEnterpris
 			}
 		}
 
-		return this.Messages.findVisibleByRoomId({ rid, latest, oldest, excludeTypes, queryOptions, inclusive, snippeted, mentionsUsername }).toArray();
+		const cursor = this.Messages.findVisibleByRoomId({ rid, latest, oldest, excludeTypes, queryOptions, inclusive, snippeted, mentionsUsername });
+		const count = cursor.count();
+		const messages = cursor.toArray();
+		Object.defineProperty(messages, 'count', { value: count });
```

Comment:
This is a nice solution,  but it would not work when dealing with micro services and encoding the data to transmit, this property will be ignored.
For this case, IMO, the best option is to return an object with 2 properties, `records` and `count`, rather than an array.